File: classical-components-demo/dc-scanner/src/main/java/io/scanbot/example/DCScannerActivity.java
package io.scanbot.example;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Toast;

import net.doo.snap.camera.CameraOpenCallback;
import net.doo.snap.camera.ScanbotCameraView;
import net.doo.snap.dcscanner.DCScanner;
import net.doo.snap.dcscanner.DCScannerFrameHandler;
import net.doo.snap.util.log.Logger;
import net.doo.snap.util.log.LoggerProvider;

import androidx.appcompat.app.AppCompatActivity;
import androidx.core.view.WindowCompat;
import io.scanbot.dcscanner.model.DisabilityCertificateRecognizerResultInfo;
import io.scanbot.sdk.ScanbotSDK;

public class DCScannerActivity extends AppCompatActivity {

    private final Logger logger = LoggerProvider.getLogger();

    private ScanbotCameraView cameraView;

    boolean flashEnabled = false;

    public static Intent newIntent(Context context) {
        return new Intent(context, DCScannerActivity.class);
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {

        supportRequestWindowFeature(WindowCompat.FEATURE_ACTION_BAR_OVERLAY);
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_dc_scanner);

        getSupportActionBar().hide();

        cameraView = (ScanbotCameraView) findViewById(R.id.camera);

        cameraView.setCameraOpenCallback(new CameraOpenCallback() {
            @Override
            public void onCameraOpened() {
                cameraView.postDelayed(new Runnable() {
                    @Override
                    public void run() {
                        cameraView.useFlash(flashEnabled);
                        cameraView.continuousFocus();
                    }
                }, 700);
            }
        });

        ScanbotSDK scanbotSDK = new ScanbotSDK(this);
        final DCScanner dcScanner = scanbotSDK.dcScanner();
        DCScannerFrameHandler dcScannerFrameHandler = DCScannerFrameHandler.attach(cameraView, dcScanner);

        dcScannerFrameHandler.addResultHandler(new DCScannerFrameHandler.ResultHandler() {
            @Override
            public boolean handleResult(DisabilityCertificateRecognizerResultInfo resultInfo) {
                if (resultInfo != null && resultInfo.recognitionSuccessful) {
                    long a = System.currentTimeMillis();

                    try {
                        startActivity(DCResultActivity.newIntent(DCScannerActivity.this, resultInfo));
                    } finally {
                        long b = System.currentTimeMillis();
                        logger.d("DCScanner", "Total scanning (sec): " + (b - a) / 1000f);
                    }
                }
                return false;
            }
        });

        findViewById(R.id.flash).setOnClickListener(new View.OnClickListener() {

            @Override
            public void onClick(View v) {
                flashEnabled = !flashEnabled;
                cameraView.useFlash(flashEnabled);
            }
        });

        Toast.makeText(
                this,
                scanbotSDK.isLicenseActive()
                        ? "License is active"
                        : "License is expired",
                Toast.LENGTH_LONG
        ).show();
    }

    @Override
    protected void onResume() {
        super.onResume();
        cameraView.onResume();
    }

    @Override
    protected void onPause() {
        super.onPause();
        cameraView.onPause();
    }
}

File: classical-components-demo/dc-scanner/src/main/java/io/scanbot/example/DCScannerActivity.kt
package io.scanbot.example

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.WindowCompat
import io.scanbot.example.DCResultActivity.Companion.newIntent
import io.scanbot.sdk.ScanbotSDK
import io.scanbot.sdk.camera.FrameHandlerResult
import io.scanbot.sdk.camera.ScanbotCameraView
import io.scanbot.sdk.dcscanner.DCScannerFrameHandler

class DCScannerActivity : AppCompatActivity() {
    private lateinit var cameraView: ScanbotCameraView

    private var flashEnabled = false

    override fun onCreate(savedInstanceState: Bundle?) {
        supportRequestWindowFeature(WindowCompat.FEATURE_ACTION_BAR_OVERLAY)
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_dc_scanner)
        supportActionBar!!.hide()

        cameraView = findViewById<View>(R.id.camera) as ScanbotCameraView
        cameraView.setCameraOpenCallback {
            cameraView.postDelayed({
                cameraView.useFlash(flashEnabled)
                cameraView.continuousFocus()
            }, 700)
        }

        val scanbotSDK = ScanbotSDK(this)
        val dcScanner = scanbotSDK.createDcScanner()
        val dcScannerFrameHandler = DCScannerFrameHandler.attach(cameraView, dcScanner)

        dcScannerFrameHandler.addResultHandler { result ->
            if (result is FrameHandlerResult.Success) {
                val resultInfo = result.value
                if (resultInfo.recognitionSuccessful) {
                    startActivity(newIntent(this, resultInfo))
                }
            }
            false
        }
        findViewById<View>(R.id.flash).setOnClickListener { v: View? ->
            flashEnabled = !flashEnabled
            cameraView.useFlash(flashEnabled)
        }
        Toast.makeText(
                this,
                if (scanbotSDK.licenseInfo.isValid) "License is active" else "License is expired",
                Toast.LENGTH_LONG
        ).show()
    }

    override fun onResume() {
        super.onResume()
        cameraView.onResume()
    }

    override fun onPause() {
        super.onPause()
        cameraView.onPause()
    }

    companion object {
        @JvmStatic
        fun newIntent(context: Context?): Intent {
            return Intent(context, DCScannerActivity::class.java)
        }
    }
}